src/query/sql/src/executor/physical_plans/physical_table_scan.rs

```diff
@@ -234,6 +239,33 @@ impl PhysicalPlanBuilder {
                 self.dry_run,
             )
             .await?;
+        // If enable block level table sample, filter parts by probability.
+        if let Some(sample) = &scan.sample {
+            match sample.sample_level {
+                SampleLevel::ROW => {}
+                SampleLevel::BLOCK => {
+                    let probability = scan.sample_probability(&None)?;
+                    if let Some(probability) = probability {
+                        let original_parts = source.parts.partitions.len();
+                        let mut sample_parts = Vec::with_capacity(original_parts);
+                        let mut rng = thread_rng();
+                        let bernoulli = Bernoulli::new(probability).unwrap();
+                        for part in source.parts.partitions.iter() {
+                            if bernoulli.sample(&mut rng) {
```

Comment:
part may be lazy, let's add a sample function into part trait.

Reply:
ok